packages/runtime/datastore-definitions/src/channel.ts

```diff
@@ -70,6 +70,8 @@ export interface IDeltaHandler {
      * @param localOpMetadata - The local metadata associated with the original message.
      */
     reSubmit(message: any, localOpMetadata: unknown): void;
+
+    rebaseOp(message: any, localOpMetadata: unknown): void;
```

Comment:
>rebaseOp [](start = 4, length = 8)

what's the difference between rebase and resubmit, seems like they could potentially be unified

Reply:
They are basically opposites. Resubmit will resubmit an op that was already submitted, rebase will cause the DDS to generate whatever local state it would have on submitting but doesn't submit (so it can be resubmitted later).